text/0000-remove-shell-processes.md

```diff
@@ -0,0 +1,401 @@
+
+## API compat
+
+### Overriding Args for Shell Processes
+We should convert args in `launch.toml` to `command` in `metadata.toml` to expose a consistent interface to users. For example, if a buildpack the following process type in `launch.toml` 
+```
+[[process]]
+type = "hi"
+command = "echo"
+args = ["hello", "world"]
+direct = true
+```
+
+we would convert it to the following in `metadata.toml`
+```
+[[process]]
+type = "hi"
+command = ["echo", "hello", "world"]
+args = []
+api = "0.5"
+direct = true
+```
+This spares users from having to learn about the differences between buildpack APIs in order to predict how additional arguments will behave.
+
+### Evaluating Args
+
+For shell processes provided by older buildpacks, we must continue to Bash evaluate buildpack-provided args to avoid breaking older buildpacks
+
+For example, if a buildpack using API `0.5` creates the following entry in `launch.toml`
+```
+[[process]]
+type = "hi"
+command = "echo"
+args = ["hello", "${WORLD:-world}"]
+```
+
+It should be converted to the following in `metadata.toml`
+
+```
+[[process]]
+type = "hi"
+command = ["echo", "hello", "${WORLD:-world}"]
+args = []
+api = "0.5"
+direct = false
+```
+
+and the launcher should Bash evaluate each entry in `command` to avoid breaking changes.
+
+However, any additional user-provided args should NOT be Bash evaluated to reduce the amount of complexity exposed to end users.
+
+### profiles
+Profiles contributed by older buildpacks will still be evaluated when executing shell process types. But user provided profiles will not be evaluated when using the new platform API, even when a shell process is executed. Again, this is done to prevent differences between the buildpack APIs from leaking into the user interface. Users will only need think about differences between platform APIs.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+In exchange for a reduction in complexity and cognitive overhead buildpack-authors and end users lose certain conveniences like the more intuitive profile interface. This drawback could be remediated by tooling for buildpack authors (to make creation of `exec.d` helpers easy) or by buildpacks (to support `.profile`). However, in the `.profile` case, consistency across ecosystems will be a matter of convention (like Procfile) rather than a guarantee.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## `$(<env>)` syntax
+
+We could use `$(<env>)` for environment variable replacements instead of `${<env>}`. urrently, Kubernetes only replaces `$(FOO)` if `FOO` matches the name of an enivornment variable defined in the `env` section of the PodSpec, otherwise `$(FOO)` is passed through literally. The would allow thelauncher to pick up where kubernetes leaves off and evaluate an remaining references.
```

Comment:
```suggestion
We could use `$(<env>)` for environment variable replacements instead of `${<env>}`. Currently, Kubernetes only replaces `$(FOO)` if `FOO` matches the name of an environment variable defined in the `env` section of the PodSpec, otherwise `$(FOO)` is passed through literally. The would allow the launcher to pick up where kubernetes leaves off and evaluate any remaining references.
```